plugins/sip/lib/model/providers/pexip/pexipUtils.php

```diff
@@ -0,0 +1,310 @@
+<?php
+/**
+ * @package plugins.sip
+ * @subpackage model.pexip
+ */
+
+class PexipUtils
+{
+	/**
+	 * @return bool
+	 * @throws Exception
+	 * @throws kCoreException
+	 */
+	public static function initAndValidateConfig()
+	{
+		if (kConf::hasMap('sip') && $pexipConfig = kConf::get('pexip', 'sip'))
+		{
+			return $pexipConfig;
+		}
+
+		KalturaLog::debug("Pexip misconfigured. please validate configuration.");
+		return false;
+	}
+
+	/**
+	 * @param $dbLiveEntry
+	 * @param $pexipConfig
+	 * @param $regenerate
+	 * @return string
+	 */
+	public static function createSipUrl(LiveStreamEntry $dbLiveEntry, $pexipConfig, $regenerate = false)
+	{
+		if (!$dbLiveEntry->getSipToken() || $regenerate)
+		{
+			$addition = str_pad(substr((string)microtime(true)*10000, -5),5,'0',STR_PAD_LEFT);
+			$sipToken = $dbLiveEntry->getPartnerId() . $addition;
+			$dbLiveEntry->setSipToken($sipToken);
+			$dbLiveEntry->save();
+		}
+		$sipUrl = $dbLiveEntry->getSipToken() . "@" . $pexipConfig['hostUrl'];
+		return $sipUrl;
+	}
+	/**
+	 * @param $entry
+	 * @param $pexipConfig
+	 * @return string
+	 */
+	public static function getRoomName(LiveEntry $entry, $pexipConfig)
+	{
+		return $entry->getId() . "@" . $pexipConfig['hostUrl'];
+	}
+
+	/**
+	 * @param $entryId
+	 * @return entry
+	 * @throws KalturaAPIException
+	 */
+	public static function validateAndRetrieveEntry($entryId)
+	{
+		$dbEntry = entryPeer::retrieveByPK($entryId);
+		if (!$dbEntry)
+		{
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
+		}
+
+		if ($dbEntry->getType() != entryType::LIVE_STREAM)
+		{
+			throw new KalturaAPIException(KalturaErrors::INVALID_ENTRY_TYPE, $dbEntry->getName(), $dbEntry->getType(), entryType::LIVE_STREAM);
+		}
+
+		return $dbEntry;
+	}
+
+	/**
+	 * @param $queryParams
+	 * @param $pexipConfig
+	 * @return bool|entry
+	 */
+	public static function retrieveAndValidateEntryForSipCall($queryParams, $pexipConfig)
+	{
+		$sipToken = self::extractSipTokenFromAddress($queryParams, $pexipConfig);
+		if (!$sipToken)
+		{
+			return false;
+		}
+
+		$partnerId = substr($sipToken, 0, -5);
+		kCurrentContext::$partner_id = $partnerId;
+		entryPeer::setUseCriteriaFilter ( false );
+		$c = KalturaCriteria::create(entryPeer::OM_CLASS);
+		$criterion = $c->getNewCriterion(entryPeer::PARTNER_ID, $partnerId);
+		$c->addAnd($criterion);
+		$c->addAnd($c->getNewCriterion(entryPeer::SIP_TOKEN, $sipToken,KalturaCriteria::EQUAL));
+		$dbLiveEntry = entryPeer::doSelectOne($c);
+		entryPeer::setUseCriteriaFilter ( true );
+		if (!$dbLiveEntry)
+		{
+			KalturaLog::err("Entry was not found for int_id $sipToken");
+			return false;
+		}
+
+		if (!PermissionPeer::isValidForPartner(PermissionName::FEATURE_SIP, $dbLiveEntry->getPartnerId()))
+		{
+			KalturaLog::err("Sip Feature is not enabled for partner " . $dbLiveEntry->getPartnerId());
```

Comment:
All the following errors should be put inti exception , so the caller can have notion of the issue, the ::err just write it internally to the logs.